### PR TITLE
#14 fixed Joomla Coding Standard and added function dock blocks

### PR DESCRIPTION
**Pull request by RonakParmar**
Pull Request for Issue # .
This PR is related with #14 
#### Summary of Changes

Fixed Joomla coding standards and added function docs blocks.
#### Testing Instructions

Install joomla coding standards in your text-editor and check file in it.

### CODE REVIEW

File: administrator/components/com_fields/models/field.php [hunk 1]
```diff
@@ -10,22 +10,47 @@
...
 	public $typeAlias = null;
 
 	private $valueCache = array();
 
+	/**
+	 * Constructor.
+	 *
+	 * @param   array  $config  An optional associative array of configuration settings.
+	 *
+	 * @since   1.6
```
Comment (zero-24) [inline administrator/components/com_fields/models/field.php]: Should be since 3.6 ;)

File: administrator/components/com_fields/models/field.php [hunk 2]
```diff
@@ -118,40 +165,70 @@ public function save ($data)
...
 	}
 
+	/**
+	 * Returns a Table object, always creating it.
+	 *
+	 * @param   string  $type    The table type to instantiate
+	 * @param   string  $prefix  A prefix for the table class name. Optional.
+	 * @param   array   $config  Configuration array for model. Optional.
+	 *
+	 * @return  JTable    A database object
```
Comment (zero-24) [inline administrator/components/com_fields/models/field.php:211]: Sine is missing

**Comment by zero-24** [comment]
Can you have a look into the since tags? I the thebmost of them should be 3.6 ;)

**Comment by laoneo** [comment]
The release will be 3.7 and not 3.6. There are still some fixes open according to travis, https://travis-ci.org/joomla-projects/custom-fields/jobs/129346825#L1311 and below.

Anyway, thanks for your ongoing help fixing the code style!

**Comment by RonakParmar** [comment]
@laoneo  I am fixing coding standards and dock blocks in a one file in a one PR.
I have replaced 3.6 by 3.7 in all since tag in this PR.

File: administrator/components/com_fields/models/field.php [hunk 3]
```diff
@@ -606,12 +794,24 @@ protected function generateNewTitle ($category_id, $alias, $title)
 		);
 	}
 
+	/**
+	 * [loadTypeForms description]
```
Comment (wilsonge) [inline administrator/components/com_fields/models/field.php]: This doc block needs fixing :)

File: administrator/components/com_fields/models/field.php [hunk 4]
```diff
@@ -400,37 +541,45 @@ public function getFieldValue ($fieldId, $context, $itemId)
...
 	}
 
 	/**
 	 * Setting the value for the gven field id, context and item id.
 	 *
-	 * @param string $fieldId
-	 * @param string $context
-	 * @param string $itemId
-	 * @param string $value
+	 * @param   string  $fieldId  [description]
```
Comment (wilsonge) [inline administrator/components/com_fields/models/field.php]: Same with the descriptions here

File: administrator/components/com_fields/models/field.php [hunk 5]
```diff
@@ -383,14 +520,18 @@ protected function preprocessForm (JForm $form, $data, $group = 'content')
 	/**
 	 * Returning the value for the given field id, context and item id.
 	 *
-	 * @param string $fieldId
-	 * @param string $context
-	 * @param string $itemId
+	 * @param   string  $fieldId  [description]
```
Comment (wilsonge) [inline administrator/components/com_fields/models/field.php]: and here

**Comment by RonakParmar** [comment]
@wilsonge I don't know the exact description of these variables, So, I keep it as it is.
If someone has idea, please write in comment, I'll change in PR.

**Comment by laoneo** [comment]
> @param string $fieldId

The ID of the field

> @param string $context

The context

> @param string $itemId

The ID of the item

Does that help?

**Comment by RonakParmar** [comment]
If it is correct description then, I'll update this PR.
I think for @param string $value description is :  The value of field .
Let me know, then I'll update this PR.

**Comment by RonakParmar** [comment]
Is there any issue in this PR? Please highlight it, so I can fix it.

File: administrator/components/com_fields/models/field.php [hunk 6]
```diff
@@ -335,6 +456,19 @@ protected function loadFormData ()
...
+	 *
+	 * Note. Calling getState in this method will result in recursion.
+	 *
+	 * @param   JForm   $form   The form object
+	 * @param   array   $data   The data to be merged into the form object
+	 * @param   string  $group  The plugin group to be executed
+	 *
+	 * @return  void
+	 *
+	 * @since    3.7
```
Comment (zero-24) [inline administrator/components/com_fields/models/field.php:468]: One space to much ;)

File: administrator/components/com_fields/models/field.php [hunk 7]
```diff
@@ -383,14 +520,18 @@ protected function preprocessForm (JForm $form, $data, $group = 'content')
...
 	 * Returning the value for the given field id, context and item id.
 	 *
-	 * @param string $fieldId
-	 * @param string $context
-	 * @param string $itemId
+	 * @param   string  $fieldId  The ID of the field
+	 * @param   string  $context  The context
+	 * @param   string  $itemId   The ID of the item
+	 *
 	 * @return NULL|string
```
Comment (zero-24) [inline administrator/components/com_fields/models/field.php]: Here we need one more space

File: administrator/components/com_fields/models/field.php [hunk 8]
```diff
@@ -400,37 +541,45 @@ public function getFieldValue ($fieldId, $context, $itemId)
...
-	 * @param string $fieldId
-	 * @param string $context
-	 * @param string $itemId
-	 * @param string $value
+	 * @param   string  $fieldId  The ID of the field
+	 * @param   string  $context  The context
+	 * @param   string  $itemId   The ID of the item
+	 * @param   string  $value    The value of the field
+	 *
 	 * @return boolean
```
Comment (zero-24) [inline administrator/components/com_fields/models/field.php]: Same here please add one space more ;)

**Comment by laoneo** [comment]
Can you update the branch, that I can merge it.

**Comment by laoneo** [comment]
I fixed all code style errors with commit 191495887caad7e73af104860197a1a14d0882a2. Can you please review it. If there are some improvements, feel free to open new pull requests. Travis should not fail anymore (https://travis-ci.org/joomla-projects/custom-fields/builds/138731999).

Thanks for all the help from your side, really appreciated.

**Pull request closed without merging**